src/components/scenes/CoinRankingDetailsScene.tsx

```diff
@@ -218,6 +271,176 @@ const CoinRankingDetailsSceneComponent = (props: Props) => {
     return rows
   }
 
+  const getWallet = async () => {
+    if (edgeAsset == null) return
+    const { pluginId, tokenId } = edgeAsset
+
+    // Find all wallets that can hold this asset
+    const matchingWallets = Object.values(account.currencyWallets).filter(wallet => wallet.currencyInfo.pluginId === pluginId)
+    if (matchingWallets.length === 0) return
+
+    // If multiple wallets, show picker
+    let targetWallet = matchingWallets[0]
+    if (matchingWallets.length > 1) {
+      const result = await Airship.show<WalletListResult>(bridge => (
+        <WalletListModal
+          bridge={bridge}
+          navigation={navigation as NavigationBase}
+          headerTitle={lstrings.select_wallet_to_send_from}
+          allowedAssets={[{ pluginId, tokenId }]}
+          showCreateWallet
+        />
+      ))
+      if (result?.type !== 'wallet') return
+      targetWallet = account.currencyWallets[result.walletId]
+    }
+
+    // If no wallet exists, create one
+    if (targetWallet == null) {
+      const walletName = getUniqueWalletName(account, pluginId)
+      targetWallet = await createWallet(account, {
+        name: walletName,
+        walletType: `wallet:${pluginId}`,
+        fiatCurrencyCode: fiatCurrencyCode
+      })
+    }
+
+    return targetWallet
+  }
+
+  const handleBuyPress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const wallet = await getWallet()
+    if (wallet == null) return
+
+    navigation.navigate('edgeTabs', {
+      screen: 'buyTab',
+      params: {
+        screen: 'pluginListBuy',
+        params: {
+          forcedWalletResult: {
+            type: 'wallet',
+            walletId: wallet.id,
+            tokenId: edgeAsset.tokenId
+          }
+        }
+      }
+    })
+  })
+
+  const handleSellPress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const wallet = await getWallet()
+    if (wallet == null) return
+
+    navigation.navigate('edgeTabs', {
+      screen: 'sellTab',
+      params: {
+        screen: 'pluginListSell',
+        params: {
+          forcedWalletResult: {
+            type: 'wallet',
+            walletId: wallet.id,
+            tokenId: edgeAsset.tokenId
+          }
+        }
+      }
+    })
+  })
+
+  const handleTradePress = useHandler(async () => {
+    if (edgeAsset == null) return
+    const { tokenId } = edgeAsset
+
+    const targetWallet = await getWallet()
+    if (targetWallet == null) return
+
+    // Find the wallet with highest USD value to use as source
+    // TODO: Include tokenIds in the sort
```

Comment:
Are we going to address this TODO, or is it out of scope for the PR?

Reply:
I wanted to address this as a fixup